share/root.go

```diff
@@ -9,6 +15,30 @@ import (
 // In practice, it is a commitment to all the Data in a square.
 type Root = da.DataAvailabilityHeader
 
+// DataHash is a representation of the Root hash.
+type DataHash []byte
+
+func (dh DataHash) Validate() error {
+	if len(dh) != 32 {
+		return fmt.Errorf("invalid hash size, expected 32, got %d", len(dh))
+	}
+	return nil
+}
+
+func (dh DataHash) String() string {
+	return fmt.Sprintf("%X", []byte(dh))
+}
+
+// IsEmptyRoot check whether DataHash corresponds to the root of an empty block EDS.
+func (dh DataHash) IsEmptyRoot() bool {
+	return bytes.Equal(EmptyRoot().Hash(), dh)
+}
+
+// NewSHA256Hasher returns a new instance of a SHA-256 hasher.
+func NewSHA256Hasher() hash.Hash {
+	return sha256.New()
+}
```

Comment:
Is this used?

Reply:
Yes, it is used everywhere instead of sha256.New(). The only place missing is latest refactoring of shwap types. I'll fix in next commit to this PR.